pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/IngestionUtils.java

```diff
@@ -88,20 +87,23 @@ private IngestionUtils() {
    */
   public static SegmentGeneratorConfig generateSegmentGeneratorConfig(TableConfig tableConfig, Schema schema)
       throws IOException, ClassNotFoundException {
-    return generateSegmentGeneratorConfig(tableConfig, schema, Collections.emptyMap());
+    Preconditions.checkNotNull(tableConfig.getIngestionConfig(),
+        "Must provide ingestionConfig in tableConfig for table: %s, for generating SegmentGeneratorConfig",
+        tableConfig.getTableName());
+    Preconditions.checkNotNull(tableConfig.getIngestionConfig().getBatchIngestionConfig(),
```

Comment:
nit: this check might be saved, as it's checked in the method call L96?

Reply:
yeah i was trying to differentiate between tableconfig that contains the batch ingestion config. vs the batchingestionconfig is individually passed in as a separate argument. but you are right no need to separately check programmatically. this is more for logging purpose.